File: app/src/main/java/com/example/weight_tracker_enrique_gonzalez/MainActivity.java
package com.example.weight_tracker_enrique_gonzalez;

import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    private EditText usernameEditText;
    private EditText passwordEditText;
    private Button loginButton;
    private Button createAccountButton;
    private Button settingsButton; // Declared the settings button

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main); // Make sure this matches your XML file name

        usernameEditText = findViewById(R.id.username);
        passwordEditText = findViewById(R.id.password);
        loginButton = findViewById(R.id.btnLogin);
        createAccountButton = findViewById(R.id.btnCreateAccount);
        settingsButton = findViewById(R.id.btnSettings); // Found the settings button by its ID

        loginButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String username = usernameEditText.getText().toString().trim();
                String password = passwordEditText.getText().toString().trim();

                if (TextUtils.isEmpty(username) || TextUtils.isEmpty(password)) {
                    Toast.makeText(MainActivity.this, "Please enter both username and password.", Toast.LENGTH_SHORT).show();
                } else {
                    Intent intent = new Intent(MainActivity.this, WeightTrackerActivity.class);
                    startActivity(intent);
                }
            }
        });

        createAccountButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String username = usernameEditText.getText().toString().trim();
                String password = passwordEditText.getText().toString().trim();

                if (TextUtils.isEmpty(username) || TextUtils.isEmpty(password)) {
                    Toast.makeText(MainActivity.this, "Please fill in all fields to create an account.", Toast.LENGTH_SHORT).show();
                } else {
                    Intent intent = new Intent(MainActivity.this, WeightTrackerActivity.class);
                    startActivity(intent);
                }
            }
        });

        // Set the click listener for the Settings button
        settingsButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                // Create an Intent to navigate to the SettingsActivity
                Intent intent = new Intent(MainActivity.this, SettingsSmsActivity.class);
                startActivity(intent);
            }
        });
    }
}

File: Enhancement/app/src/main/java/com/example/weight_tracker/MainActivity.kt
package com.example.weight_tracker

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)

        val emailInput = findViewById<EditText>(R.id.emailInput)
        val passwordInput = findViewById<EditText>(R.id.passwordInput)
        val loginBtn = findViewById<Button>(R.id.loginBtn)
        val createAccountBtn = findViewById<Button>(R.id.btnCreateAccount)

        val prefs = getSharedPreferences("UserPrefs", MODE_PRIVATE)

        // Login
        loginBtn.setOnClickListener {
            val email = emailInput.text.toString().trim()
            val pass = passwordInput.text.toString().trim()

            if (email.isEmpty() || pass.isEmpty()) {
                Toast.makeText(this, "Please fill in all fields", Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }

            // Look for a password saved under this specific email key
            val savedPass = prefs.getString(email, null)

            if (savedPass != null && savedPass == pass) {
                // Mark as logged in and save which user is logged in
                prefs.edit().apply {
                    putBoolean("isLoggedIn", true)
                    putString("lastLoggedInUser", email)
                    apply()
                }
                navigateToHome()
            } else {
                Toast.makeText(this, "Invalid email or password", Toast.LENGTH_SHORT).show()
            }
        }

        // Creat account
        createAccountBtn.setOnClickListener {
            val email = emailInput.text.toString().trim()
            val pass = passwordInput.text.toString().trim()

            // Validate Email format
            if (!android.util.Patterns.EMAIL_ADDRESS.matcher(email).matches()) {
                Toast.makeText(this, "Please enter a valid email", Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }

            if (pass.length < 4) {
                Toast.makeText(this, "Password must be at least 4 characters", Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }

            // Check if user already exists
            if (prefs.contains(email)) {
                Toast.makeText(this, "User already exists! Try logging in.", Toast.LENGTH_SHORT).show()
            } else {
                // Save new user; email = key, pass = value
                prefs.edit().putString(email, pass).apply()
                Toast.makeText(this, "Account Created! You can now log in.", Toast.LENGTH_SHORT).show()

                // Clear fields
                emailInput.text.clear()
                passwordInput.text.clear()
            }
        }

        // If already logged in, skip this screen
        if (prefs.getBoolean("isLoggedIn", false)) {
            navigateToHome()
        }
    }

    private fun navigateToHome() {
        val intent = Intent(this, HomeActivity::class.java)
        startActivity(intent)
        finish()
    }
}